core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
> Ah, the original problem... I think it's that INSTR_RAW_BITS_VALID gets cleared by the use of instr_get_opcode, perhaps by a call to instr_reuse.

Looks like decode_opcode should set the raw bits before returning NULL on an invalid instr (and probably that CLIENT_ASSERT should not be there since it's called internally like this).  Messy -- but so is the IT block tracking so I don't think there's a simpler solution.

Why is it OP_UNDECODED in the first place -- ARM doesn't have a fast decoder so isn't the opcode always set or set to OP_INVALID?